pkg/reconciler/taskrun/taskrun_test.go

```diff
@@ -1707,7 +1707,335 @@ spec:
+		expectedTr:              canceledTaskRun,
+		expectCompletionTimeSet: true,
+	}, {
+		name:       "Retry on TimedOut",
+		task:       simpleTask,
+		tr:         toBeTimedOutTaskRun,
+		expectedTr: timedOutTaskRun,
+	}, {
+		name:       "Retry on TaskRun Pod Failure",
+		task:       simpleTask,
+		tr:         toFailOnPodFailureTaskRun,
+		expectedTr: failedOnPodFailureTaskRun,
+		pod:        failedPod,
+	}, {
+		name:           "Retry on TaskRun Prepare Failure",
+		tr:             toFailOnPrepareTaskRun,
+		expectedTr:     failedOnPrepareTaskRun,
+		reconcileError: prepareError,
+	}, {
+		name:           "Retry on TaskRun Reconciliation Failure",
+		task:           resultsTask,
+		tr:             toFailOnReconcileFailureTaskRun,
+		expectedTr:     failedOnReconcileFailureTaskRun,
+		reconcileError: reconciliatonError,
+	}} {
+		t.Run(tc.name, func(t *testing.T) {
+			d := test.Data{
+				TaskRuns: []*v1beta1.TaskRun{tc.tr},
+				ConfigMaps: []*corev1.ConfigMap{{
+					ObjectMeta: metav1.ObjectMeta{Namespace: system.Namespace(), Name: config.GetFeatureFlagsConfigName()},
+					Data: map[string]string{
+						"enable-api-fields": config.AlphaAPIFields,
+					},
+				}},
+			}
+			if tc.task != nil {
+				d.Tasks = []*v1beta1.Task{tc.task}
+			}
+			if tc.pod != nil {
+				d.Pods = []*corev1.Pod{tc.pod}
+			}
+			testAssets, cancel := getTaskRunController(t, d)
+			defer cancel()
+			c := testAssets.Controller
+			clients := testAssets.Clients
+			createServiceAccount(t, testAssets, "default", tc.tr.Namespace)
+
+			err := c.Reconciler.Reconcile(testAssets.Ctx, getRunName(tc.tr))
+			if tc.reconcileError != nil {
+				if err == nil {
+					t.Error("expect error in TaskRun reconciliation, but got nil")
+				} else {
+					if d := cmp.Diff(strings.TrimSuffix(err.Error(), "\n\n"), tc.reconcileError.Error()); d != "" {
+						t.Errorf("Expected: %v, but Got: %v", tc.reconcileError, err)
+					}
+				}
+			} else if err != nil {
+				if ok, _ := controller.IsRequeueKey(err); !ok && tc.reconcileError == nil {
+					t.Errorf("unexpected error in TaskRun reconciliation: %v", err)
+				}
+			}
+
+			reconciledTaskRun, err := clients.Pipeline.TektonV1beta1().TaskRuns("foo").Get(testAssets.Ctx, tc.tr.Name, metav1.GetOptions{})
+			if err != nil {
+				t.Fatalf("Somehow had error getting reconciled run out of fake client: %s", err)
+			}
+
+			ignoreCompletionTime := cmpopts.IgnoreFields(v1beta1.TaskRunStatusFields{}, "CompletionTime")
+			ignoreObjectMeta := cmpopts.IgnoreFields(metav1.ObjectMeta{}, "Labels", "ResourceVersion", "Annotations")
+			ignoreStatusTaskSpec := cmpopts.IgnoreFields(v1beta1.TaskRunStatusFields{}, "TaskSpec")
+			ignoreTaskRunStatusFields := cmpopts.IgnoreFields(v1beta1.TaskRunStatusFields{}, "Steps", "Sidecars")
+			if d := cmp.Diff(reconciledTaskRun, tc.expectedTr, ignoreLastTransitionTime, ignoreObjectMeta, ignoreCompletionTime, ignoreStatusTaskSpec, ignoreTaskRunStatusFields); d != "" {
```

Comment:
consider grouping all the fields into a `[]ignore` slice and then expanding the slice as the third value to `cmp.Diff`?

Reply:
SG, updated.